### ISSUE TITLE
Cannot execute spoon from Java 9 Project

### ISSUE THREAD

**Issue by fsomme2s** [INRIA/spoon#1599]
I had a project running with java 8, updated it to java 9 and now spoon crashes while building the model:

```java
  Launcher launcher = new Launcher();
  // ... launcher config
  launcher.buildModel(); // *Crash*
```

While this works in a project built with java 8, when I compile and run the same project with jdk 9, I get the following error:

```
Exception in thread "main" spoon.reflect.declaration.ParentNotInitializedException: parent not initialized for (class spoon.support.reflect.reference.CtExecutableReferenceImpl)
	at spoon.support.reflect.declaration.CtElementImpl.getParent(CtElementImpl.java:317)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:29)
	at spoon.reflect.visitor.CtScanner.visitCtInvocation(CtScanner.java:479)
	at spoon.support.reflect.code.CtInvocationImpl.accept(CtInvocationImpl.java:57)
...
```

I did neither change the code that starts the launcher, nor did I change the code that the launcher analyzes to build the model. So the only change I did was to compile and run my project that depends on spoon with Java 9.

**Comment by pvojtechovsky** [comment]
Could you append whole spoon stacktrace so it is visible in which part of process it failed?

> updated it to java 9 

what exactly did you updated?
A) compile with Java 8, but running with Java 9 runtime
B) compile with Java 9, and running with Java 9 runtime
C) compile with Java 9, and running with Java 9 runtime and change source compatibility to java 9
D) ?

It would be also nice if you might provide failing test case.

**Comment by fsomme2s** [comment]
Hi, thanks for the quick response :) I was not so fast, sorry! ;)

To Answer your question: C) is what I did, I think... Well, I was working within my IDE (intelliJ IDEA) so I hope there was no additional "background magic" - but as far as I can tell, I did change the "source level" to Java 9, used the jdk9/bin/javac for compiling, and the jdk9/bin/java for running the result.

I created a small project for demonstration: https://github.com/fsomme2s/spoon-java9-test 
While doing this, **I found the Java class that caused the problem in the code analyzed by spoon: It is a subclass of java.lang.RuntimeException, that overrides a constructor method**. 

In the example project, it is placed within the directory: "src/main/resources/input-src/com/example" and is used as input for Spoon.

Here the full stack trace: 

```
Exception in thread "main" spoon.reflect.declaration.ParentNotInitializedException: parent not initialized for (class spoon.support.reflect.reference.CtExecutableReferenceImpl)
	at spoon.support.reflect.declaration.CtElementImpl.getParent(CtElementImpl.java:317)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:29)
	at spoon.reflect.visitor.CtScanner.visitCtInvocation(CtScanner.java:479)
	at spoon.support.reflect.code.CtInvocationImpl.accept(CtInvocationImpl.java:57)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:139)
	at spoon.reflect.visitor.CtScanner.visitCtBlock(CtScanner.java:272)
	at spoon.support.reflect.code.CtBlockImpl.accept(CtBlockImpl.java:49)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.visitCtConstructor(CtScanner.java:340)
	at spoon.support.reflect.declaration.CtConstructorImpl.accept(CtConstructorImpl.java:58)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:139)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:308)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:69)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:139)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:629)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:57)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:139)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:628)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:57)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:139)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:628)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:57)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.checkModel(JDTBasedSpoonCompiler.java:139)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:133)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:107)
	at spoon.Launcher.buildModel(Launcher.java:705)
	at spoon.java9.test.app.App.main(App.java:17)
```

**Comment by surli** [comment]
Hi @fsomme2s 

I just have a look on your project. The project is linked with the usage of an older version of JDT which is not compliant with JDK9. Especially when it concerns core elements of the language.
I'm currently trying to port Spoon to use JDK9 and the new version of JDT (see #1585) but I cannot give you right now an idea of when it'll be ready :-/

**Comment by fsomme2s** [comment]
Ah ok - so basically it's work in progress and all I need to do is to wait for the fixed version? I'm fine with that! :)

**Comment by monperrus** [comment]
closed per #1585
